core/commands/IssueUpdateCommand.php

```diff
@@ -0,0 +1,621 @@
+			) ) || ( $this->updated_issue->resolution < $t_resolution_fixed_threshold && $this->updated_issue->status >= $t_resolved_status
+			)
+			) ) {
+			throw new ClientException(
+				sprintf( 'Resolution "%1$s" is not allowed for status "%2$s".', get_enum_element( 'resolution', $this->updated_issue->resolution ), get_enum_element( 'status', $this->updated_issue->status ) ), 
+				ERROR_INVALID_RESOLUTION, 
+				array( 
+					get_enum_element( 'resolution', $this->updated_issue->resolution ), 
+					get_enum_element( 'status', $this->updated_issue->status ), 
+				) );
+		}
+
+		# Ensure that the user has permission to change the target version of the issue.
+		if( $this->existing_issue->target_version !== $this->updated_issue->target_version ) {
+			if( !access_has_bug_level( config_get( 'roadmap_update_threshold' ), $t_issue_id, null ) ) {
+				throw new ClientException( 
+					'Access Denied.', 
+					ERROR_ACCESS_DENIED );
+			}
+		}
+
+		# Ensure that the user has permission to change the view status of the issue.
+		if( $this->existing_issue->view_state != $this->updated_issue->view_state ) {
+			if( !access_has_bug_level( config_get( 'change_view_status_threshold' ), $t_issue_id, null ) ) {
+				throw new ClientException( 
+					'Access Denied.', 
+					ERROR_ACCESS_DENIED );
+			}
+		}
+
+		# Ensure that the user has permission to change the sticky of the issue.
+		if( $this->existing_issue->sticky != $this->updated_issue->sticky ) {
+			if( !access_has_bug_level( config_get( 'set_bug_sticky_threshold' ), $t_issue_id, null ) ) {
+				throw new ClientException( 
+					'Access Denied.', 
+					ERROR_ACCESS_DENIED );
+			}
+		}
+
+		mci_project_custom_fields_validate( $this->updated_issue->project_id, $this->custom_fields_to_set );
+
+		# Perform validation of the duplicate ID of the bug.
+		if( $this->updated_issue->duplicate_id != 0 ) {
+			if( $this->updated_issue->duplicate_id == $t_issue_id ) {
+				throw new ClientException( 
+					'You cannot set an issue as a duplicate of itself.', 
+					ERROR_BUG_DUPLICATE_SELF );
+			}
+
+			bug_ensure_exists( $this->updated_issue->duplicate_id );
+
+			if( !access_has_bug_level( config_get( 'update_bug_threshold' ), $this->updated_issue->duplicate_id ) ) {
+				throw new ClientException( 
+					sprintf( 'Access denied: The issue %1$d requires higher access level.', $this->updated_issue->duplicate_id ), 
+					ERROR_RELATIONSHIP_ACCESS_LEVEL_TO_DEST_BUG_TOO_LOW, 
+					array( $this->updated_issue->duplicate_id ) );
+			}
+		}
+
+		if( isset( $t_notes ) && is_array( $t_notes ) ) {
+			$t_bugnotes = bugnote_get_all_visible_bugnotes( $t_issue_id, 'DESC', 0 );
+
+			foreach( $t_bugnotes as $t_bugnote ) {
+				$this->existing_notes[$t_bugnote->id] = $t_bugnote;
+			}
+
+			$t_reassign_on_feedback_trigger = false;
+
+			foreach( $t_notes as $t_note ) {
+				$t_note = ApiObjectFactory::objectToArray( $t_note );
+
+				if( isset( $t_note['view_state'] ) ) {
```

Comment:
Default view state is not specified should only be done for new notes, for existing notes, they view_state should remain as is.